libsolidity/interface/OptimiserSettings.h

```diff
@@ -0,0 +1,72 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @author Alex Beregszaszi
+ * @date 2017
+ * Helper class for optimiser settings.
+ */
+
+#pragma once
+
+namespace dev
+{
+namespace solidity
+{
+
+struct OptimiserSettings
+{
+	/// Reset to defaults.
+	void reset()
+	{
+		runOrderLiterals = false;
+		runPeephole = false;
```

Comment:
Selected each and every option via StandardCompiler is simple but we need to come up with a way to do it in the CLI.

And supporting presets (none, basic, all) is probably a good idea, but we need to define what they contain.